contracts/src/systems/labor/build_labor.cairo

```diff
@@ -196,97 +196,75 @@ mod tests {
-        Serde::serialize(@100_000, ref mint_stone_calldata); // amount
-        world.execute('MintResources', mint_stone_calldata);
+        Serde::serialize(
+            @21_000_000_000_000_000_000, ref create_labor_conf_calldata
+        ); // base_food_per_cycle
+        world.execute('SetLaborConfig', create_labor_conf_calldata);
 
         // set block timestamp in order to harvest labor
         // initial ts = 0
         let ts = 10_000;
         starknet::testing::set_block_timestamp(ts);
 
+        let labor_resource_type = get_labor_resource_type(resource_type);
+
+        // switch to executor to set storage directly
+        starknet::testing::set_contract_address(world.executor());
+        set!(
+            world,
+            Resource {
+                entity_id: realm_entity_id.try_into().unwrap(),
+                resource_type: labor_resource_type,
+                balance: 40
+            }
+        );
+        starknet::testing::set_contract_address(player_address);
+
         // build labor for gold
         let mut build_labor_calldata = array![];
         Serde::serialize(@realm_entity_id, ref build_labor_calldata); // realm_id
-        Serde::serialize(@ResourceTypes::GOLD, ref build_labor_calldata); // resource_type
+        Serde::serialize(@resource_type, ref build_labor_calldata); // resource_type
         Serde::serialize(@20, ref build_labor_calldata); // labor_units
         Serde::serialize(@1, ref build_labor_calldata); // multiplier
         world.execute('BuildLabor', build_labor_calldata);
 
-        // assert resources are the right amount
-        let coal_resource = get!(world, (realm_entity_id, ResourceTypes::COAL), Resource);
-        assert(coal_resource.resource_type == ResourceTypes::COAL, 'failed resource type');
-        assert(coal_resource.balance == 80_000, 'failed resource amount');
-
-
-        let stone_resource = get!(world, (realm_entity_id, ResourceTypes::STONE), Resource);
-        assert(stone_resource.resource_type == ResourceTypes::STONE, 'failed resource type');
-        assert(stone_resource.balance == 80_000, 'failed resource amount');
-
         // assert labor is right amount
-        let gold_labor = get!(world, (realm_entity_id, ResourceTypes::GOLD), Labor);
+        let gold_labor = get!(world, (realm_entity_id, resource_type), Labor);
         assert(gold_labor.balance == ts + (7_200 * 20), 'wrong gold labor balance');
         assert(gold_labor.last_harvest == ts, 'wrong gold labor last harvest');
         assert(gold_labor.multiplier == 1, 'wrong gold multiplier');
 
+        let gold_resource_type = get!(world, (realm_entity_id, labor_resource_type), Resource);
+        assert(gold_resource_type.balance == 20, 'wrong labor resource');
     }
 
 
-
     #[test]
     #[available_gas(300000000000)]
     fn test_build_labor_food() {
         let world = spawn_eternum();
 
+        let player_address = starknet::get_caller_address();
```

Comment:
Technically, the player_address is the test contract address. That's why we do `starknet::testing::set_contract_address` and not `starknet::testing::set_caller_address` before calling `world.execute`